test/realm_set_test.dart

```diff
@@ -0,0 +1,535 @@
+      final testSet = TestRealmSets(1);
+      final set = testSet.setByType(type).set;
+      final values = testSet.values(type);
+
+      set.add(values.first);
+      expect(set.length, equals(1));
+      expect(set.contains(values.first), true);
+
+      set.remove(values.first);
+      expect(set.length, equals(0));
+      expect(set.contains(values.first), false);
+    });
+
+    test('RealmSet<$type> unmanged set remove', () {
+      final testSet = TestRealmSets(1);
+      final set = testSet.setByType(type).set;
+      final values = testSet.values(type);
+
+      set.add(values.first);
+      expect(set.length, equals(1));
+      expect(set.contains(values.first), true);
+
+      set.remove(values.first);
+      expect(set.length, equals(0));
+      expect(set.contains(values.first), false);
+    });
+
+    test('RealmSet<$type> unmanged set elementAt', () {
+      final testSet = TestRealmSets(1);
+      final set = testSet.setByType(type).set;
+      final values = testSet.values(type);
+
+      set.add(values.first);
+      expect(set.length, equals(1));
+      expect(set.elementAt(0), values.first);
+    });
+
+    test('RealmSet<$type> create', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      var testSet = TestRealmSets(1);
+
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      expect(realm.find<TestRealmSets>(1), isNotNull);
+
+      testSet = realm.find<TestRealmSets>(1)!;
+      var set = testSet.setByType(type).set;
+
+      expect(set.length, equals(0));
+    });
+
+    test('RealmSet<$type> create from unmanaged', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      var testSet = TestRealmSets(1);
+      var set = testSet.setByType(type).set;
+      var values = testSet.values(type);
+
+      for (var value in values) {
+        set.add(value);
+      }
+
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      testSet = realm.find<TestRealmSets>(1)!;
+      set = testSet.setByType(type).set;
+      expect(set.length, equals(values.length));
+      values = testSet.getValuesOrManagedValues(realm, type);
+
+      for (var value in values) {
+        expect(set.contains(value), true);
+      }
+    });
+
+    test('RealmSet<$type> contains', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      var testSet = TestRealmSets(1);
+      var set = testSet.setByType(type).set;
+      var values = testSet.values(type);
+
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      set = testSet.setByType(type).set;
+
+      expect(set.contains(values.first), false);
+
+      realm.write(() {
+        set.add(values.first);
+      });
+
+      testSet = realm.find<TestRealmSets>(1)!;
+      set = testSet.setByType(type).set;
```

Comment:
This shouldn't be needed. We already have the right `set` instance and it is refreshed.